src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs

```diff
@@ -182,6 +177,11 @@ protected virtual string GetAssemblyName()
             return PathUtilities.GetFileName(assemblyName);
         }
 
+        protected bool IsProjectReferenceOutputAssembly(MSB.Framework.ITaskItem item)
+        {
+            return item.GetMetadata("ReferenceOutputAssembly") == "true";
```

Comment:
Possibly that doesn't flow into the references.  An alternative here might be to look at the `_ResolvedProjectReferencePaths` item group instead and remove them from the rest of the references